modules/preview-react/mutliselect/MultiSelectInput.tsx

```diff
@@ -0,0 +1,306 @@
+    // });
+  }
+};
+
+const useMultiSelectedItem = composeHooks(
+  createElemPropsHook(useListModel)((model, ref, elemProps) => {
+    return {
+      onKeyDown(event: React.KeyboardEvent<HTMLElement>) {
+        const id = event.currentTarget.dataset.id || '';
+        if (event.key === 'Backspace' || event.key === 'Delete') {
+          model.events.select({id});
+          removeItem(id, model);
+        }
+      },
+      onClick(event: React.MouseEvent<HTMLElement>) {
+        const id = event.currentTarget.dataset.id || '';
+        model.events.select({id});
+      },
+    };
+  }),
+  useListItemRovingFocus,
+  useListItemRegister
+);
+
+const MultiSelectedItem = createSubcomponent('span')({
+  modelHook: useListModel,
+  elemPropsHook: useMultiSelectedItem,
+})(({children, ref, ...elemProps}, Element, model) => {
+  return (
+    <Pill as={Element} variant="removable">
+      {children}
+      <Pill.IconButton ref={ref} {...(elemProps as any)} />
+    </Pill>
+  );
+});
+
+export interface MultiSelectInputProps
+  extends CSProps,
+    Pick<
+      React.InputHTMLAttributes<HTMLInputElement>,
+      'disabled' | 'className' | 'style' | 'aria-labelledby'
+    > {}
+
+export const MultiSelectInput = createSubcomponent(TextInput)({
+  displayName: 'MultiSelect.Input',
+  modelHook: useMultiSelectModel,
+  elemPropsHook: useMultiSelectInput,
+})<MultiSelectInputProps>(
+  (
+    {className, cs, style, 'aria-labelledby': ariaLabelledBy, formInputProps, ...elemProps},
+    Element,
+    model
+  ) => {
+    return (
+      <div {...handleCsProp({className, cs, style}, multiSelectStencil({}))}>
+        <InputGroup>
+          <InputGroup.Input data-part="form-input" {...formInputProps} />
+          <InputGroup.Input
+            data-part="user-input"
+            as={Element}
+            aria-labelledby={ariaLabelledBy}
+            readOnly
+            {...elemProps}
+          />
+          <InputGroup.InnerEnd pointerEvents="none">
+            <SystemIcon icon={caretDownSmallIcon} />
+          </InputGroup.InnerEnd>
+        </InputGroup>
+        {model.selected.state.items.length ? (
+          <>
+            <div data-part="separator" />
+            <ListBox
+              model={model.selected}
+              as="div"
+              role="listbox"
+              aria-orientation="horizontal"
+              aria-labelledby={ariaLabelledBy}
+            >
+              {item => <MultiSelectedItem>{item.textValue}</MultiSelectedItem>}
+            </ListBox>
+          </>
+        ) : null}
+      </div>
+    );
+  }
+);
+
+export const MultiSelectSearchInput = createSubcomponent(TextInput)({
+  displayName: 'MultiSelect.Input',
+  modelHook: useMultiSelectModel,
+  elemPropsHook: useMultiSelectInput,
+})<MultiSelectInputProps>(
+  ({className, cs, 'aria-labelledby': ariaLabelledBy, ...elemProps}, Element, model) => {
+    return (
+      <div {...handleCsProp({className, cs}, multiSelectStencil({}))}>
```

Comment:
does this order matter where you pass `cs` ?

Reply:
I'm not sure what this means. We're passing an object to `handleCsProp` which destructs the arguments. In general, when passing an object to a function, the key order should not matter.